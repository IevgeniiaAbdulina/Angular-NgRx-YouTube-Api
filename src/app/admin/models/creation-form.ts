export interface CustomVideoFormFields {
  title: CustomFormField;
  description: CustomFormField;
  imageCoverLink: CustomFormField;
  videoLink: CustomFormField;
  creationDate: CustomFormField;
  tags: CustomFormField;
}

export interface CustomFormField {
  name: string;
  type: string;
  label: string;
  formControlName: string;
  mapedErrors: Map<string, string> | null;
  errorKeys: string[];
}
