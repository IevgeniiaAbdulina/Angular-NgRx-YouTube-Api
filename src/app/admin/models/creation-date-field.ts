import { CustomFormField } from './creation-form';

export class CreationDateField implements CustomFormField {
  name = 'creation-date';
  type = 'date';
  label = 'Creation date';
  formControlName = 'creationDate';
  mapedErrors = new Map([['required', 'Please enter a creation date']]);
  errorKeys: string[];

  constructor() {
    this.errorKeys = [...this.mapedErrors.keys()];
  }
}
