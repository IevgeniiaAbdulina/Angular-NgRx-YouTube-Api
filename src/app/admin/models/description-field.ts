import { CustomFormField } from './creation-form';

export class DescriptionField implements CustomFormField {
  name = 'description';
  type = 'text';
  label = 'Description';
  formControlName = 'description';
  mapedErrors = new Map([['maxlength', 'The description is too long']]);
  errorKeys: string[];

  constructor() {
    this.errorKeys = [...this.mapedErrors.keys()];
  }
}
