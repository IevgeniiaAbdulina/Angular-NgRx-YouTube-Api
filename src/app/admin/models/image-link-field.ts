import { CustomFormField } from './creation-form';

export class ImageCoverLinkField implements CustomFormField {
  name = 'image-cover';
  type = 'url';
  label = 'Image Cover Link';
  formControlName = 'imageCoverLink';
  mapedErrors = new Map([['required', 'Please enter a link to the image']]);
  errorKeys: string[];

  constructor() {
    this.errorKeys = [...this.mapedErrors.keys()];
  }
}
