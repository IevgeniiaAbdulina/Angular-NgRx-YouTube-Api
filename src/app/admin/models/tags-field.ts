import { CustomFormField } from './creation-form';

export class TagsField implements CustomFormField {
  name = 'tags';
  type = 'text';
  label = 'Tag:';
  formControlName = 'tags';
  mapedErrors = new Map([['required', 'Please enter a tag']]);
  errorKeys: string[];

  constructor() {
    this.errorKeys = [...this.mapedErrors.keys()];
  }
}
