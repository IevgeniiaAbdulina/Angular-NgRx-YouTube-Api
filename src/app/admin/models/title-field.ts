import { CustomFormField } from './creation-form';

export class TitleField implements CustomFormField {
  name = 'title';
  type = 'text';
  label = 'Title';
  formControlName = 'title';
  mapedErrors = new Map([
    ['required', 'Please enter a title'],
    ['minlength', 'The title is too short'],
    ['maxlength', 'The title is too long']
  ]);
  errorKeys: string[];

  constructor() {
    this.errorKeys = [...this.mapedErrors.keys()];
  }
}
