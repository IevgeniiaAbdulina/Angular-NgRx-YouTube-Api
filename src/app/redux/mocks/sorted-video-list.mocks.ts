import { VideoItem } from 'src/app/shared/models/video-list-response';

export const sortedVideoListByDateMock: VideoItem[] = [
  {
    kind: 'youtube#searchResult',
    etag: 'j7Nz5_A6m-rRRF3wv35-gfA80rk',
    id: 'CSU7SggjVwg',
    snippet: {
      publishedAt: '2023-12-04T20:45:41Z',
      channelId: 'UCPgzHeVeVDsUvcvovWHjHGQ',
      title:
        'Różaniec Teobańkologia z prorokiem Izajaszem o górach i przepaściach 4.12 Poniedziałek',
      description:
        'Różaniec Teobańkologia odmawiamy codziennie na żywo o 20:30. Dziś rozważamy tajemnice radosne. Rozważania różańcowe ...',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/CSU7SggjVwg/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '1',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  },
  {
    kind: 'youtube#searchResult',
    etag: '5788M5Le5PZfiXxLQkkKfLxxJYo',
    id: '_iyVsT2s2xg',
    snippet: {
      publishedAt: '2023-12-04T20:37:00Z',
      channelId: 'UCDfk8ogO6QGeJAYCY0QDzKw',
      title: 'My TESA Autopilot Failed!!',
      description:
        'My brand new Tesla auto pilot failed. Just like the cyber truck. My Tesla is brand brand new and it has auto pilot full self driving.',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/_iyVsT2s2xg/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '1',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  }
];

export const videoListSortedByViewsMock: VideoItem[] = [
  {
    kind: 'youtube#searchResult',
    etag: 'RfFF-gdDOPOms7GUgui1Rdf2mFY',
    id: 'YbQVrJXfuJM',
    snippet: {
      publishedAt: '2018-05-24T12:29:37Z',
      channelId: 'UC5ljgk-ZnRBrC4fnbSGVsDQ',
      title: 'BLUE BIRD | cały film | PL | short love movie',
      description:
        'Film powstały we współpracy z Warszawską Szkołą Filmową i Polskim Instytutem Sztuki Filmowej. KONTAKT: ...',
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/YbQVrJXfuJM/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '100',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  },
  {
    kind: 'youtube#searchResult',
    etag: '3CKhG34WA4J9DiWR5fkxGBJ3Fe8',
    id: 'N-PyWfVkjZc',
    snippet: {
      publishedAt: '2020-11-26T17:00:11Z',
      channelId: 'UCsfdrdj6DAWmBUdPDfFTatg',
      title:
        'Shakin&#39; Stevens - Merry Christmas Everyone (Official 4K Video)',
      description:
        "Shakin' Stevens - Merry Christmas Everyone | Stream Shakin' Stevens here: https://shakinstevens.lnk.to/MCE2022ID Listen to all ...",
      thumbnails: {
        high: {
          url: 'https://i.ytimg.com/vi/N-PyWfVkjZc/hqdefault.jpg',
          width: 480,
          height: 360
        }
      },
      channelTitle: 'testing',
      tags: ['1'],
      categoryId: '',
      liveBroadcastContent: '',
      defaultLanguage: '',
      localized: {
        title: '',
        description: ''
      },
      defaultAudioLanguage: ''
    },
    statistics: {
      viewCount: '50',
      likeCount: '2',
      dislikeCount: '3',
      favoriteCount: '4',
      commentCount: '5'
    }
  }
];
