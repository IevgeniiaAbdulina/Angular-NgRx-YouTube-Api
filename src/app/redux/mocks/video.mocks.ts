import { CustomVideo } from 'src/app/shared/models/custom-video-interface';
import { VideoItem } from 'src/app/shared/models/video-list-response';

import { CustomVideoState, VideoState } from '../state.models';

export const videoItemMock: VideoItem = {
  id: '123',
  snippet: {
    publishedAt: '12/2/2023',
    channelId: '',
    title: 'First mock video item',
    description: 'Test video item',
    thumbnails: {
      high: {
        url: 'assets/img/default_image.svg',
        width: 100,
        height: 150
      }
    },
    channelTitle: 'testing',
    tags: ['1'],
    categoryId: '',
    liveBroadcastContent: '',
    defaultLanguage: '',
    localized: {
      title: '',
      description: ''
    },
    defaultAudioLanguage: ''
  },
  statistics: {
    viewCount: '1',
    likeCount: '2',
    dislikeCount: '3',
    favoriteCount: '4',
    commentCount: '5'
  },
  kind: 'mockYoutube#video',
  etag: '1'
};

export const videoListMock: VideoItem[] = [videoItemMock, videoItemMock];

export const favoriteListMock: VideoItem[] = [
  videoItemMock,
  videoItemMock,
  videoItemMock
];

export const customVideoItemMock: CustomVideo = {
  id: '456',
  snippet: {
    publishedAt: '12/2/2023',
    channelId: '',
    title: 'First mock video item',
    description: 'Test video item',
    thumbnails: {
      high: {
        url: 'assets/img/default_image.svg',
        width: 100,
        height: 150
      }
    },
    channelTitle: 'testing',
    tags: ['1'],
    categoryId: '',
    liveBroadcastContent: '',
    defaultLanguage: '',
    localized: {
      title: '',
      description: ''
    },
    defaultAudioLanguage: ''
  },
  statistics: {
    viewCount: '1',
    likeCount: '2',
    dislikeCount: '3',
    favoriteCount: '4',
    commentCount: '5'
  },
  kind: 'mockCustom#video',
  etag: '1'
};

export const customVideoListMock: CustomVideo[] = [
  customVideoItemMock,
  customVideoItemMock
];

export const videoListStateMock: VideoState = {
  isLoading: false,
  videoList: videoListMock,
  favoriteList: videoListMock,
  error: ''
};

export const customVideoStateMock: CustomVideoState = {
  isLoading: false,
  customVideoList: customVideoListMock,
  error: ''
};
