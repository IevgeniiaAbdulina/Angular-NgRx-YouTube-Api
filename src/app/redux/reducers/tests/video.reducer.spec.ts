import {
  addToFavoriteList,
  getFavoriteList,
  getFavoriteListSuccess,
  getVideoList,
  getVideoListFailure,
  getVideoListSuccess,
  removeFromFavoriteList,
  setLoading
} from '../../actions/video-api.actions';
import { videoItemMock, videoListMock } from '../../mocks/video.mocks';
import { VideoState } from '../../state.models';
import { initialSate, reducers } from '../video.reducer';

describe('Video Reducer', () => {
  let initialVideoState: VideoState;

  beforeEach(() => {
    initialVideoState = { ...initialSate };
  });

  it('should set isLoading state to true on loading page first time', () => {
    const result = reducers(initialSate, setLoading({ isLoading: true }));

    expect(result.isLoading).toEqual(true);
    expect(result).toMatchSnapshot();
  });

  it('should set loading flag to false on getVideoList action', () => {
    const result = reducers(initialVideoState, getVideoList());

    expect(result).toEqual({
      isLoading: false,
      videoList: [],
      favoriteList: [],
      error: null
    });
  });

  it('should save received data to store and set loading flag to false on getVideoListSuccess action', () => {
    const result = reducers(
      initialSate,
      getVideoListSuccess({
        videoList: videoListMock
      })
    );

    expect(result).toEqual({
      isLoading: false,
      videoList: videoListMock,
      favoriteList: [],
      error: null
    });
  });

  it('[Video List] Get Video List Failure', () => {
    const error = 'error';
    const action = getVideoListFailure({ error });

    const expectedState: VideoState = {
      ...initialSate,
      error
    };

    const result = reducers(initialSate, action);

    expect(result).toEqual(expectedState);
  });

  it('should change state when getFavoriteList', () => {
    const result = reducers(initialSate, getFavoriteList());

    expect(result).toEqual({
      isLoading: true,
      videoList: [],
      favoriteList: [],
      error: null
    });
  });

  it('should change state when getFavoriteListSuccess', () => {
    const result = reducers(
      initialSate,
      getFavoriteListSuccess({
        favoriteList: videoListMock
      })
    );

    expect(result).toEqual({
      isLoading: false,
      videoList: [],
      favoriteList: videoListMock,
      error: null
    });
  });

  describe('[Add Favorite] Add Favorite Video', () => {
    it('should change state when item has been added to favorite video list', () => {
      const result = reducers(
        initialSate,
        addToFavoriteList({
          favoriteItem: videoItemMock
        })
      );

      expect(result).toEqual({
        ...initialSate,
        favoriteList: [videoItemMock]
      });
    });
  });

  describe('[Remove Favorite] Remove Favorite Video', () => {
    it('should change state when item has been removed from favorite video list', () => {
      const result = reducers(
        initialSate,
        removeFromFavoriteList({
          favoriteItem: videoItemMock
        })
      );

      expect(result).toEqual({
        ...initialSate,
        favoriteList: []
      });
    });
  });
});
