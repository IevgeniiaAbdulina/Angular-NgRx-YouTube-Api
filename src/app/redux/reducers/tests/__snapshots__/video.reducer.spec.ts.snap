// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Reducer should set loading state 1`] = `
{
  "error": null,
  "favoriteList": [],
  "isLoading": true,
  "videoList": [],
}
`;

exports[`videoReducer should set loading state 1`] = `
{
  "error": null,
  "favoriteList": [],
  "isLoading": true,
  "videoList": [],
}
`;

exports[`videoreducer should set loading state 1`] = `
{
  "error": null,
  "favoriteList": [],
  "isLoading": true,
  "videoList": [],
}
`;
